cli/azd/pkg/infra/provisioning/bicep/prompt.go

```diff
@@ -96,8 +162,40 @@ func (p *BicepProvider) promptForParameter(
 		azdMetadata.Type != nil &&
 		*azdMetadata.Type == azure.AzdMetadataTypeLocation {
 
+		// location can be combined with allowedValues and with usageName metadata
+		// allowedValues == nil => all locations are allowed
+		// allowedValues != nil => only the locations in the allowedValues are allowed
+		// usageName != nil => the usageName is validated for quota for each allowed location (this is for Ai models),
+		//                     reducing the allowed locations to only those that have quota available
+		// usageName == nil => No quota validation is done
+		var allowedLocations []string
+		if param.AllowedValues != nil {
+			allowedLocations = make([]string, len(*param.AllowedValues))
+			for i, option := range *param.AllowedValues {
+				allowedLocations[i] = option.(string)
+			}
+		}
+		if azdMetadata.UsageName != nil {
+			if allowedLocations == nil {
```

Comment:
Should we also consider if `allowedLocations` is empty, but not nil?

Reply:
When allowedLocations is empty, the template author should notice the template fails b/c the initial prompt would return error after no options to show